### PR TITLE
Content deletion isn't propagated to a grandchild layer #9922

### PR DESCRIPTION
**Pull request by vbradnitski**

### CODE REVIEW

File: modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/DeletedEventSyncCommand.java [hunk 1]
```diff
@@ -21,37 +33,92 @@ public static Builder create()
...
+
+    private boolean hasNoChildren( final ContentToSync contentToSync, final Set<ContentId> idsToRemove )
+    {
+        return !contentToSync.getTargetContent().hasChildren() ||
+            getAllChildren( contentToSync.getTargetContent().getId() ).stream().allMatch( idsToRemove::contains );
+    }
+
+    private boolean hasNoInboundDependencies( final ContentToSync contentToSync, final Set<ContentId> idsToRemove )
+    {
+        return getInboundDependencies( contentToSync.getTargetContent().getId() ).stream().allMatch( idsToRemove::contains );
```
Comment (rymsha) [inline modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/DeletedEventSyncCommand.java:98]: Optimize - move to query

File: modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ParentContentSynchronizer.java [hunk 2]
```diff
@@ -246,14 +258,38 @@ else if ( actualTargetContext != null )
...
             }
+            else
+            {
+                return null;
+            }
+
+            if ( targetContent != null && ( targetContent.getOriginProject() == null ||
+                !actualSourceContext.getRepositoryId().equals( targetContent.getOriginProject().getRepoId() ) ) )
+            {
+                return null; // do not sync contents without origin parent or from other source
```
Comment (rymsha) [inline modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ParentContentSynchronizer.java:269]: This is a tricky part and must be covered with test

File: modules/runtime/src/bin/server.sh [hunk 3]
```diff
@@ -37,7 +37,7 @@ locateJava() {
 }
 
 setupDefaults() {
-    DEFAULT_JAVA_OPTS="-XX:-OmitStackTraceInFastThrow -XX:+AlwaysPreTouch"
+    DEFAULT_JAVA_OPTS="-XX:-OmitStackTraceInFastThrow -XX:+AlwaysPreTouch -XX:-MaxFDLimit"
```
Comment (rymsha) [inline modules/runtime/src/bin/server.sh:40]: Let's have a separate issue for this

File: modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/DeletedEventSyncCommand.java (hunk 1 above)
Comment (vbradnitski) [reply modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/DeletedEventSyncCommand.java:98]: 'idsToRemove' conversation could be really big

File: modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ParentContentSynchronizer.java (hunk 2 above)
Comment (vbradnitski) [reply modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ParentContentSynchronizer.java]: removed it, it's unreal behavior outside of multi-inheritance POC for now

**Review by rymsha — APPROVED** [review]

**Review by anatol-sialitski — APPROVED** [review]

**Pull request merged**